### PR TITLE
fix: Shutdown Connect when migrations fail

### PR DESCRIPTION
**Pull request by filipecabaco**
## What kind of change does this PR introduce?
Currently we had an issue where Connect module was not shutting down properly leading to more and more connections to be created against the tenant database. Now we use reason :shutdown to enforce the process is killed.

Also the existing Listen/Notify approach will be deleted for a different approach to be built in the future.

### CODE REVIEW

File: lib/realtime/tenants/connect.ex [hunk 1]
```diff
@@ -97,37 +95,21 @@ defmodule Realtime.Tenants.Connect do
     Logger.metadata(external_id: tenant_id, project: tenant_id)
 
     with %Tenant{} = tenant <- Tenants.get_tenant_by_external_id(tenant_id),
-         res <- Database.check_tenant_connection(tenant, @application_name),
+         {:ok, conn} <- Database.check_tenant_connection(tenant, @application_name),
+         ref = Process.monitor(conn),
```
Comment (abc3) [inline lib/realtime/tenants/connect.ex:99]: Is it worth using link() instead of monitor()?

**Review by abc3 — APPROVED** [review]
LGTM

File: lib/realtime/tenants/connect.ex (hunk 1 above)
Comment (filipecabaco) [reply lib/realtime/tenants/connect.ex:99]: We're using monitor here to ensure that we track the connection shutdown and handle it to also kill the Connect process

**Pull request merged**